nservicebus/how-do-i-handle-responses-on-the-client-side.md

```diff
@@ -17,18 +17,25 @@ public class MyHandler : IHandleMessages<MyMessage>
 }
 ```
 
-You will probably need to access other objects in the client process in this handler. Use [dependency injection](containers.md) , declaring a property or constructor argument of the necessary type in your message handler. Then, register the relevant object in the container (most likely as a singleton).
-
 ## Handling responses in the context of the request
 
 When sending a message/request, you can register a callback that is invoked when a response arrives. This callback is invoked before any message handler as it is registered like this:
 
 
 ```C#
-bus.Send( messageInstance ).Register( asyncCallback, state );
+bus.Send(messageInstance).Register(asyncCallback, state);
 ```
 
-DANGER: If the server process returns multiple responses, NServiceBus cannot know which response message will be the last. To prevent memory leaks, the callback is invoked only for the first response. Callbacks won't survive a crash as they are held in memory, so they are less suitable for server-side development where fault-tolerance is required. In those cases, [sagas are preferred](sagas-in-nservicebus.md) .
+DANGER: If the server process returns multiple responses, NServiceBus cannot know which response message will be the last. To prevent memory leaks, the callback is invoked only for the first response. Callbacks won't survive a crash as they are held in memory, so they are less suitable for server-side development where fault-tolerance is required. In those cases, [sagas are preferred](sagas-in-nservicebus.md).
+
+To trigger a callback, you need to return an `enum` or `int` value.
+
+<!-- import TriggerCallback -->
+
+## Handling callbacks in the context of web application
+
+Callbacks can be used to provide an acknowledgement for successfully dispatched command, but should not be used instead of message handlers. Message handlers should handle asynchronous communication using [Full Duplex](/samples/fullduplex). Request/response with callback is possible, although discouraged, see reason outlined above. 

```

Comment:
I have removed mentioning of `RegisterWebCallback` as it was removed as of v3.
As for the rest of the comment, completely makes sense. Will update.